src/test/fuzz/txdownloadman_one_honest_peer.cpp

```diff
@@ -0,0 +1,522 @@
+{
+    TestPeer peer;
+    peer.m_nodeid = nodeid;
+    peer.m_good = force_good || fuzzed_data_provider.ConsumeBool();
+    peer.m_connection_preferred = preferred;
+    peer.m_relay_permissions = fuzzed_data_provider.ConsumeBool();
+    // good peer should be at least WTXID_RELAY_VERSION
+    peer.m_wtxid_relay = force_good || fuzzed_data_provider.ConsumeBool();
+    return peer;
+}
+
+static TxValidationResult TESTED_TX_RESULTS[] = {
+    // This can lead to orphan resolution
+    TxValidationResult::TX_MISSING_INPUTS,
+    // Reconsiderable failures - can lead to package validation.
+    TxValidationResult::TX_RECONSIDERABLE,
+    // Represents all non-reconsiderable failures - it's not necessary to distinguish between policy
+    // and consensus here.
+    TxValidationResult::TX_MEMPOOL_POLICY,
+};
+
+
+const TestingSetup* g_setup;
+constexpr size_t NUM_COINS{100};
+// map from uint256 (can be txid or wtxid) to 2 transactions of the same txid.
+// If the key is a wtxid, the first tx in the pair has that wtxid, and the second doesn't.
+std::map<uint256, std::pair<CTransactionRef, CTransactionRef>> TRANSACTIONS;
+
+static CTransactionRef MakeTransactionSpending(const std::vector<COutPoint>& outpoints, size_t num_outputs, bool add_witness)
+{
+    CMutableTransaction tx;
+    for (const auto& outpoint : outpoints) {
+        tx.vin.emplace_back(outpoint);
+    }
+    if (add_witness) {
+        tx.vin[0].scriptWitness.stack.push_back({1});
+    }
+    tx.vout.emplace_back(CENT, P2WSH_OP_TRUE);
+    return MakeTransactionRef(tx);
+}
+
+static CTransactionRef Malleate(const CTransactionRef& ptx)
+{
+    CMutableTransaction tx(*ptx);
+    tx.vin[0].scriptWitness.stack.push_back({5});
+    auto mutated_tx = MakeTransactionRef(tx);
+    assert(ptx->GetHash() == mutated_tx->GetHash());
+    return mutated_tx;
+}
+
+static CTransactionRef StripWitness(const CTransactionRef& ptx)
+{
+    CMutableTransaction tx(*ptx);
+    tx.vin[0].scriptWitness.stack.clear();
+    auto stripped_tx = MakeTransactionRef(tx);
+    assert(ptx->GetHash() == stripped_tx->GetHash());
+    return stripped_tx;
+}
+
+void initialize()
+{
+    static const auto testing_setup = MakeNoLogFileContext<const TestingSetup>();
+    g_setup = testing_setup.get();
+    for (uint32_t i = 0; i < uint32_t{NUM_COINS}; ++i) {
+        COutPoint outpoint{Txid::FromUint256((HashWriter() << i).GetHash()), i};
+
+        auto correct_tx = MakeTransactionSpending({outpoint}, /*num_outputs=*/2, /*add_witness=*/true);
+        auto malleated_tx = Malleate(correct_tx);
+        auto stripped_tx = StripWitness(correct_tx);
+
+        const auto& target_txid = correct_tx->GetHash().ToUint256();
+        const auto& target_wtxid = correct_tx->GetWitnessHash().ToUint256();
+        const auto& malleated_wtxid = malleated_tx->GetWitnessHash().ToUint256();
+
+        // The first tx in the pair needs to match by wtxid.
```

Comment:
I'm not sure what this comment means. I know the `honestid` peer needs to be wtxid relaying to ensure that it can properly serve `malleated_tx` when necessary, otherwise `GetCorrectTx` will fetch the wrong response by giving back `correct_tx`.